Firmware/ultralcd.cpp

```diff
@@ -983,6 +983,15 @@ void lcd_commands()
                 lcd_commands_step = 3;
                 break;
             case 3:
+#ifdef FILAMENT_SENSOR
+                if (fsensor.isEnabled() && fsensor.getFilamentPresent()) {
+                    lcd_show_fullscreen_message_and_wait_P(_T(MSG_UNLOAD_FILAMENT_REPEAT));
+                    lcd_commands_step = 0;
+                    lcd_commands_type = LcdCommands::Idle;
+                    SetPrinterState(PrinterState::Idle);
```

Comment:
Can we re-use case 1 to avoid duplicating code? It saves a little bit of flash memory.
```suggestion
                    lcd_commands_step = 1;
```

Reply:
I didn't use it because it exits to the hardware setup menu, so you would need to manually go up two menu levels to get to the unload option.

Reply:
Fair enough. This isn't a blocker IMO.

I had another idea though I want to throw out here :)

Would it be simpler (code-wise) to check if the filament is present in the menu item instead?:

```cpp
static void nozzle_change()
{
    lcd_commands_type = LcdCommands::NozzleCNG;
    lcd_return_to_status();
}
```
Below is untested code:

```cpp
static void nozzle_change()
{
    if (fsensor.isEnabled() && fsensor.getFilamentPresent()) {
        lcd_show_fullscreen_message_and_wait_P(_T(MSG_UNLOAD_FILAMENT_REPEAT));
    } else {
        lcd_commands_type = LcdCommands::NozzleCNG;
    }
    lcd_return_to_status();
}
```

I don't remember if `lcd_show_fullscreen_message_and_wait_P` is still a problem within a menu item, but I think it will work today since there is no call to `lcd_update()`.

If this doesn't work, then please ignore my comment. It's just an idea.

Reply:
Changing it in the menu item works and saves some bytes. 👍